pkg/git/local_repository.go

```diff
@@ -326,35 +327,84 @@ type PullResult struct {
 	IsUpdated bool
 }
 
-func (localRepo *LocalRepository) Pull(auth AuthMethod) (PullResult, error) {
+func (localRepo *LocalRepository) Pull(auth AuthMethod) (*PullResult, error) {
 	var gitAuth transport.AuthMethod
 	if auth != nil {
 		gitAuth = auth.toGitAuthMethod()
 	}
+
 	var err error
+	originType := unknownOriginType
+	if !localRepo.DryRun {
+		originType, err = localRepo.originType()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	goGit := originType == httpsOriginType
+
 	log.Debug().
 		Str("repo", localRepo.Path()).
+		Bool("go_git", goGit).
 		Bool("dry_run", localRepo.DryRun).
 		Msg("git: pull")
 	if !localRepo.DryRun {
-		err = localRepo.worktree.Pull(&git.PullOptions{
-			RemoteName: OriginRemote,
-			Auth:       gitAuth,
-		})
-	}
-	if err != nil && !errors.Is(err, git.NoErrAlreadyUpToDate) {
-		return PullResult{}, err
+		if goGit {
+			err = localRepo.worktree.Pull(&git.PullOptions{
+				RemoteName: OriginRemote,
+				Auth:       gitAuth,
+			})
+			if err != nil && !errors.Is(err, git.NoErrAlreadyUpToDate) {
+				return nil, err
+			}
+		} else {
+			if stdout, stderr, err := shell.RunCommand(localRepo.Path(), "git", "pull", OriginRemote); err != nil {
+				return nil, fmt.Errorf("git pull failed:\n\tstdout:[%s]\n\tstderr:[%s]\n%s", stdout, stderr, err.Error())
+			}
+		}
+
 	}
-	return PullResult{IsUpdated: err == nil}, nil
+	return &PullResult{IsUpdated: true}, nil
```

Comment:
Why?

Reply:
there are no other return paths where err != nil, so this becomes `nil == nil` which is `true`

Reply:
in the impossible case where err != nil, shouldn't we `return PullResult{...}, *err*`

Reply:
go-git returns the error in case local repo is already up to date, so you lose this info.
By the way, can we check the same for binary git?